sycl/source/handler.cpp

```diff
@@ -288,6 +288,7 @@ event handler::finalize() {
         OutEvent = &NewEvent->getHandleRef();
 
         NewEvent->setSubmissionTime();
+        NewEvent->setQueueBaseTime();
```

Comment:
The existing submit time was captured after enqueue (kernel launching). The host side sycl timestamp second timestamp is captured earliest possible without changing the original submit time.

Reply:
Doesn't this make the queue base time essentially the same as submission time?
Submission time here captures the moment when the command is submitted to the SYCL runtime. The queue base time should reflect the point at which the command is enqueued to the underlying backend, i.e. right before we make the corresponding piEnqueue* call.

Reply:
Unless we move the NewEvent->setSubmissionTime(); position which captures host timestamp 1 in step 1, capturing host base timestamp before piEnqueue will be earlier time than submit time since submit time is captured after kernel launching. Still it would be a good idea to do that? Also it was a bit tricky to add the timestamp capturing code in handler due to the mutex. Once  in a while the event could not be completed.